vendor/brave-ios/Ledger/BATBraveLedger.mm

```diff
@@ -110,7 +109,9 @@ - (instancetype)initWithStateStoragePath:(NSString *)path
 
     ledgerClient = new NativeLedgerClient(self);
     ledger = ledger::Ledger::CreateInstance(ledgerClient);
-    ledger->Initialize();
+    ledger->Initialize(^(int32_t result){
```

Comment:
@kylehickinson In `RewardsService`, we're using the same callback as `createWallet`. Should IOS behave the same?

Reply:
Not sure what you mean? Like the contents of the callback? Or just the same type of callback (`int32_t result`)

Reply:
The contents of the callback, ex `RewardsServiceImpl` uses `OnWalletInitialized` for both `Initialize` and `CreateWallet`

Reply:
Ah I see what you mean. Basically that we want to call the wallet initialized observers. Currently there's more logic involved around creating wallets callback than just observers, so how about for now lets just makes `Initalize`'s callback do this:

```
ledger->Initialize(^(int32_t result){
  for (BATBraveLedgerObserver *observer in self.observers) {
    if (observer.walletInitalized) {
      observer.walletInitalized(static_cast<BATResult>(result));
    }
  }
});
```

And I will cleanup the implementation at a later date

Reply:
Will do, thanks for the feedback.